### PR TITLE
added calculate Neff script

### PR DESCRIPTION
**Pull request by qiyunzhu**
Hello @tkosciol @sjanssen2 I have created this code in response to issue #26 .

I am not 100% certain the calculation is correct. Please review the code with your scientific mind. Specifically, what I noticed that could be problems are:
 - Missing residues (`-`) are treated as the 21st character, which may result in the overestimation of number of clusters.
 - Clustering is performed using `scipy`'s hierarchical clustering function. Not sure if it's the desired way.

The actual code is very simple and can be compressed in five lines. The reason I created so many functions is because I guess other parts of the program may want to call some of these functions.

### CODE REVIEW

File: microprot/scripts/calculate_Neff.py [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+    -------
+    list of int
+        flat cluster numbers to which sequences are assigned
+    """
+    t = 1.0 - cutoff / 100.0
+    return list(fcluster(linkage(dm.condensed_form()), t,
+                         criterion='distance'))
+
+
+def effective_family_size(clusters, length):
```
Comment (tkosciol) [inline microprot/scripts/calculate_Neff.py]: Add a flag to return either `Neff` as defined before, or `Nclu`, not scaled by the length of the sequence. Make it return `Neff` by default.

**Comment by tkosciol** [comment]
Looks good to me!
@sjanssen2 any comments? In the interest of time I want to leave it with Hamming distance for now and I'll keep an issue open, so that we know to upgrade this script later.

**Comment by sjanssen2** [comment]
That's fine to me :thumbsup:

**Comment by qiyunzhu** [comment]
Thank you guys! @tkosciol @sjanssen2

**Pull request merged**